optaplanner-examples/src/main/java/org/optaplanner/examples/common/swingui/ConstraintMatchesDialog.java

```diff
@@ -56,6 +56,14 @@ public ConstraintMatchesDialog(SolverAndPersistenceFrame solverAndPersistenceFra
         this.solutionBusiness = solutionBusiness;
     }
 
+    @Override
+    public void setVisible(boolean visible) {
+        if (!visible) {
+            setContentPane(new JPanel()); // force redraw on visible(true)
```

Comment:
You're right, of course. I wasn't aware of `invalidate()`, only of `validate()`, and that didn't work. Fixed, will push ASAP.